### PR TITLE
Cleanup for boost 1.63.0

### PR DESCRIPTION
**Pull request by fwyzard**
Fix the errors encountered while building CMSSW 9.0.x with boost 1.63.0 .

### CODE REVIEW

**Comment by fwyzard** [comment]
please test

File: CondFormats/SiStripObjects/src/SiStripBaseDelay.cc [hunk 1]
```diff
@@ -5,6 +5,8 @@
 #include <iostream>
 #include <sstream>
 
+#include <boost/bind.hpp>
```
Comment (davidlt) [inline CondFormats/SiStripObjects/src/SiStripBaseDelay.cc]: `boost::bind` stuff can be replace with `std::bind`.

File: PhysicsTools/Utilities/interface/Primitive.h [hunk 2]
```diff
@@ -156,10 +156,8 @@ namespace funct {
   //  /
   
   template <TYPXT2,
-    bool bint = ::boost::type_traits::ice_not<
-    ::boost::is_same<PRIMIT(X, B), UndefinedIntegral>::value>::value,
-    bool aint = ::boost::type_traits::ice_not<
-    ::boost::is_same<PRIMIT(X, A), UndefinedIntegral>::value>::value>
+    bool bint = not ::boost::is_same<PRIMIT(X, B), UndefinedIntegral>::value,
+    bool aint = not ::boost::is_same<PRIMIT(X, A), UndefinedIntegral>::value>
```
Comment (davidlt) [inline PhysicsTools/Utilities/interface/Primitive.h]: `std::is_same` would again eliminate the need for boost. Same applies to the stuff bellow.

**Comment by davidlange6** [comment]
@ggovi - what would need to test  conditions compatibility with boost 1.63? is there a unit test?

File: CondFormats/SiStripObjects/src/SiStripBaseDelay.cc (hunk 1 above)
Comment (fwyzard) [inline CondFormats/SiStripObjects/src/SiStripBaseDelay.cc]: I agree, please submit a patch.

**Comment by ianna** [comment]
+1

**Comment by kpedro88** [comment]
+1

**Comment by ggovi** [comment]
@davidlange6 
yes, there is a unit test:
CondCore/CondDB/test/condTestRegression.py
currently the (hardcoded) release list for the comparison is out of date. 
I've just executed a private version executing the cross-operation (write with one and read with the previous one ) with the following list:
766 gcc493, 8026 gcc530, 810 gcc530, 900p3 gcc530. 
Any idea if this is sufficient? If yes, I do the PR for the updated list, which should go in before to validate the new boost. Otherwise, i can add more entries in the list...

**Comment by davidlange6** [comment]
Hi @ggovi
is it not possible to add the current release (eg, the IB in which the test would be run)? the releases look ok otherwise..

**Comment by ggovi** [comment]
@davidlange6 
the 'underlying' release , or IB, or private checkout area, is the target of the comparison, we don't need to add it. What we need to make sure, is that all the (already validated) releases/platforms in our production environment are represented in the comparison matrix.

**Comment by davidlange6** [comment]
ok, perfect.

Thanks

**Comment by fwyzard** [comment]
Can I suggest to use 900p4 instead of 900p3 ?

.A

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/17454	5c94507
- technical changes
- jenkins tests pass

**Pull request merged**